File: server/src/modules/event/infrastructure/repository.ts
import { IEventRepository } from '../domain/repository.interface';
import { Event } from '../domain/entity';
import { InMemoryRepository } from '../../../shared/infrastructure/in-memory.repository';
import { Injectable } from '@nestjs/common';

@Injectable()
export class InMemoryEventRepository
  extends InMemoryRepository<Event>
  implements IEventRepository
{
  entities: Event[] = [
    new Event(1, 'Event 1', 'Description 1'),
    new Event(2, 'Event 2', 'Description 2'),
    new Event(3, 'Event 3', 'Description 3'),
    new Event(4, 'Event 4', 'Description 4'),
  ];
}

File: server/src/modules/event/infrastructure/repository.py
from typing import List

from modules.event.domain.entity import Event
from modules.event.domain.repository import EventRepository
from modules.event.infrastructure.orm import Event as ORMEvent
from shared.infrastructure.repository import SQLAlchemyAsyncRepository


class EventSqlAlchemyRepository(EventRepository, SQLAlchemyAsyncRepository[ORMEvent]):
    def get_by_location(self, location: str) -> List[Event]:
        return self.get_by_filters(location=location)
